controller/forwarding/hyperlane_test.go

```diff
@@ -0,0 +1,475 @@
+					sdk.NewInt64Coin("usdn", 1),
+					[]byte{},
+				)
+				require.NoError(t, err)
+
+				return f
+			},
+			expHypAttr: forwardingtypes.HypAttributes{
+				TokenId:           []byte("token id"),
+				DestinationDomain: 0,
+				Recipient:         make([]byte, 32),
+				CustomHookId:      make([]byte, 32),
+				GasLimit:          math.NewInt(1),
+				MaxFee:            sdk.NewInt64Coin("usdn", 1),
+			},
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			controller, err := forwarding.NewHyperlaneController(
+				log.NewNopLogger(),
+				mocks.HyperlaneHandler{},
+			)
+			require.NoError(t, err)
+
+			hypAttr, err := controller.ExtractAttributes(tC.forwarding())
+
+			if tC.expError != "" {
+				require.ErrorContains(t, err, tC.expError)
+				require.Nil(t, hypAttr)
+			} else {
+				require.NoError(t, err)
+				require.NotNil(t, hypAttr)
+				require.Equal(t, tC.expHypAttr.DestinationDomain, hypAttr.DestinationDomain)
+				require.Equal(t, len(tC.expHypAttr.Recipient), len(hypAttr.Recipient))
+				require.Equal(t, len(tC.expHypAttr.CustomHookId), len(hypAttr.CustomHookId))
+				require.True(t, tC.expHypAttr.GasLimit.Equal(hypAttr.GasLimit))
+				require.Equal(t, tC.expHypAttr.MaxFee, hypAttr.MaxFee)
+			}
+		})
+	}
+}
+
+func TestValidateForwarding_Hyperlane(t *testing.T) {
+	transferAttr, err := types.NewTransferAttributes(
+		core.PROTOCOL_IBC,
+		"channel-0",
+		"usdn",
+		math.NewInt(1),
+	)
+	require.NoError(t, err)
+
+	usdnID := make([]byte, 32)
+	copy(usdnID, "usdn id")
+
+	testCases := []struct {
+		name         string
+		setup        func(*mocks.HyperlaneHandler)
+		hypAttr      *forwardingtypes.HypAttributes
+		transferAttr *types.TransferAttributes
+		expError     string
+	}{
+		{
+			name: "success - when the attributes are valid",
+			setup: func(m *mocks.HyperlaneHandler) {
+				tokenID := hyperlaneutil.HexAddress(usdnID)
+				hypToken := &warptypes.WrappedHypToken{
+					Id:            tokenID.String(),
+					Owner:         "noble",
+					TokenType:     0,
+					OriginMailbox: "",
+					OriginDenom:   "usdn",
+					IsmId:         &hyperlaneutil.HexAddress{},
+				}
+
+				m.Tokens[hypToken.Id] = *hypToken
+			},
+			transferAttr: transferAttr,
+			hypAttr: &forwardingtypes.HypAttributes{
+				TokenId:           usdnID,
+				DestinationDomain: 0,
+				Recipient:         make([]byte, 32),
+				CustomHookId:      make([]byte, 32),
+				GasLimit:          math.NewInt(1),
+				MaxFee:            sdk.NewInt64Coin("usdn", 1),
+			},
+		},
+		{
+			name: "error - when the hyperlane denom is not destination denom",
+			setup: func(m *mocks.HyperlaneHandler) {
+				tokenID := hyperlaneutil.HexAddress(usdnID)
+				hypToken := &warptypes.WrappedHypToken{
+					Id:            tokenID.String(),
+					Owner:         "noble",
+					TokenType:     0,
+					OriginMailbox: "",
+					OriginDenom:   "btc",
+					IsmId:         &hyperlaneutil.HexAddress{},
+				}
+
+				m.Tokens[hypToken.Id] = *hypToken
+			},
```

Comment:
same here, it would be nice to define this as `validHypToken` and then only set the changed field in the setup function, e.g.:

```suggestion
			setup: func(m *mocks.HyperlaneHandler) {
				hypToken := validHypToken
                hypToken.OriginDenom = "btc"

				m.Tokens[hypToken.Id] = *hypToken
			},
```

This way would be clearer (at least for me) what specifically was changed per test-case. WDYT?